### ISSUE TITLE
[Content] Hot-reloading Content at run-time

### ISSUE THREAD

**Issue by craftworkgames** [MonoGame-Extended/Monogame-Extended#554]
A while back I experimented with the idea of hot-reloading game content while the game is running. This is could be a pretty cool feature of Extended if we get it right.

The idea of hot loading isn't new. It's quite common in front-end web development these days. If you've used it I'm sure you'll know how awesome it can be. For those that haven't, the main advantage is that you can make changes to your game while playing it. It can really boost development time because you can rapidly iterate and change things until they look right.

Here are some references:
- https://vue-loader.vuejs.org/guide/hot-reload.html
- https://stackoverflow.com/questions/41428954/what-is-the-difference-between-hot-reloading-and-live-reloading-in-react-native
- https://facebook.github.io/react-native/blog/2016/03/24/introducing-hot-reloading.html
- https://www.quora.com/Whats-hot-loading-in-Webpack-and-how-does-it-work

More importantly I've got this gist with some of the code I used to experiment with this idea a while back. I'm pretty sure I got a little further than this at the time, but I must've lost that code somehow. Anyways, it will still make a good starting point.

https://gist.github.com/craftworkgames/98a9eee11be1399915fd7d6f1e20818d

**Comment by esengine** [comment]
This is really cool. I think the lua language may also do this, we can try the lua language

http://nlua.org/

**Comment by Apostolique** [comment]
I'll be keeping my eyes on this. This feature would be awesome!

**Comment by LithiumToast** [comment]
@craftworkgames You might be interested in [some code I quickly wrote](https://github.com/LithiumToast/Assetyard/tree/dirtywip/src/Styxz.Assetyard/src) when thinking of handling assets. I wish I had more time to work on getting the project going 😞

**Comment by craftworkgames** [comment]
@LithiumToast yeah thanks mate. I remember when you started that project. It's a shame you've not had time to work on it but I get it. We've all got our lives to live.

**Comment by AristurtleDev** [comment]
The new code-first content system for monogame being created as part of base MonoGame will include a content sever that will allow for hot reloading of assets.  Given that this is something that will be part of base MonoGame soon, I'm going to put this issue on hold until we see the full release and if there is any value that can be added to it through MGE at that point.
